File: solutions/javascript/move-zeros.js
function moveZeros(arr) {
  const nonZero = arr.filter(x => x !== 0);
  const zeros = arr.length - nonZero.length;
  return [...nonZero, ...Array(zeros).fill(0)];
}
console.log(moveZeros([0,1,0,3,12]));

File: solutions/cpp/move_zeros.cpp
#include <iostream>
#include <vector>
using namespace std;

vector<int> moveZeros(vector<int>& arr) {
    vector<int> r;
    int zeros = 0;
    for(int x:arr) {
        if(x==0) zeros++;
        else r.push_back(x);
    }
    while(zeros--) r.push_back(0);
    return r;
}

int main() {
    vector<int> v={0,1,0,3,12};
    auto r=moveZeros(v);
    for(int x:r) cout<<x<<" ";
}
